plugins/docker_binary_builder/app/models/binary_builder.rb

```diff
@@ -0,0 +1,136 @@
+class BinaryBuilder
+  DOCKER_BUILD_FILE = 'Dockerfile.build'.freeze
+  BUILD_SCRIPT = '/app/build.sh'.freeze
+  ARTIFACTS_FILE = 'artifacts.tar'.freeze
+  ARTIFACTS_FILE_PATH = "/app/#{ARTIFACTS_FILE}".freeze
+  DOCKER_HOST_CACHE_DIR = '/opt/samson_build_cache'.freeze
```

Comment:
@steved, this is was granted us huge improvements in the build phase for our Scala project.
Maven and SBT(ivy) take a huge amount of time to download all the dependencies. We reduced our build phase to at most 2 minutes, instead of the 15, 20 minutes.

I tried using the layering cache that docker offers by adding the build file (build.scala), first and running a fetch dependencies command. However the build file changes too often causing the cache not to work on most of the times (causing the build of an image to take 15, 20 minutes,...painful).